src/components/table/table.tsx

```diff
@@ -113,16 +128,29 @@ export function Table<T extends object>({
           [state, controlledState]
         )
       },
-      groupByFn,
+      toggleSelectedItemClb,
+      itemIsDisabled,
     },
     ...reactTableHooks
   )
 
   useEffect(() => {
-    if (selectedItemsClb) {
-      selectedItemsClb(selectedFlatRows.map((r: Row<T>) => r.original))
+    if ((selectedFlatRows.length === 0 || isAllRowsSelected) && selectedItemsClb) {
+      const isGrouped = groupBy.length > 0
+      selectedItemsClb(
+        selectedFlatRows.reduce((h: Item[], r: Row<T>) => {
```

Comment:
Do `h` and `r` represent some naming convention? We generally use `acc` and `current`/`entity name from context`.